linkerd/app/src/env.rs

```diff
@@ -488,85 +486,164 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
+                                        detect_protocol_timeout,
+                                    );
+                                    (p, allow)
+                                })
+                                .collect::<HashMap<_, _>>();
+                        if id_disabled && !ports.is_empty() {
+                            error!(
+                                "if {} is true, {} must be empty",
+                                ENV_IDENTITY_DISABLED, ENV_INBOUND_PORTS_REQUIRE_IDENTITY
+                            );
+                            return Err(EnvError::InvalidEnvVar);
+                        }
+                        ports
+                    };
+
+                    let require_tls_ports = {
+                        let mut ports =
+                            parse(strings, ENV_INBOUND_PORTS_REQUIRE_TLS, parse_port_set)?
+                                .unwrap_or_default()
+                                .into_iter()
+                                .map(|p| {
+                                    let allow = policy::defaults::all_mtls_unauthenticated(
+                                        detect_protocol_timeout,
+                                    );
+                                    (p, allow)
+                                })
+                                .collect::<HashMap<_, _>>();
+                        // `require_identity` is more restrictive than `require_tls`, so prefer it if
+                        // there are duplicates.
+                        for p in require_identity_ports.keys() {
+                            ports.remove(p);
+                        }
+                        if id_disabled && !ports.is_empty() {
+                            error!(
+                                "if {} is true, {} must be empty",
+                                ENV_IDENTITY_DISABLED, ENV_INBOUND_PORTS_REQUIRE_TLS
+                            );
+                            return Err(EnvError::InvalidEnvVar);
+                        }
+                        ports
+                    };
+
+                    let opaque_ports = {
+                        let ports = inbound_disable_ports?
+                            .unwrap_or_default()
                             .into_iter()
-                            .filter_map(|p| allow_opaque.clone().map(move |a| (p, a))),
-                    ),
-            )
+                            .filter_map(|p| match default {
+                                policy::DefaultPolicy::Allow(ref a) => {
+                                    let mut sp = (**a).clone();
+                                    sp.protocol = inbound::policy::Protocol::Opaque;
+                                    Some((p, sp))
+                                }
+                                policy::DefaultPolicy::Deny => {
+                                    tracing::warn!("inbound opaque ports configuration is ignored when the default policy is 'deny'");
```

Comment:
nit: i think `warn!` is imported in this module?
```suggestion
                                    warn!("inbound opaque ports configuration is ignored when the default policy is 'deny'");
```